kube-core/src/params.rs

```diff
@@ -140,6 +140,33 @@ impl ListParams {
     }
 }
 
+/// The validation directive to use for `fieldValidation` when using server-side apply.
+#[derive(Clone, Debug)]
+pub enum ValidationDirective {
+    /// This will fail the request with a BadRequest error if any unknown fields would be dropped from the
+    /// object, or if any duplicate fields are present. The error returned from the server will contain
+    /// all unknown and duplicate fields encountered.
+    Strict,
+    /// This will send a warning via the standard warning response header for each unknown field that
+    /// is dropped from the object, and for each duplicate field that is encountered. The request will
+    /// still succeed if there are no other errors, and will only persist the last of any duplicate fields.
+    Warn,
+    /// This will ignore any unknown fields that are silently dropped from the object, and will ignore
+    /// all but the last duplicate field that the decoder encounters.
+    Ignore,
```

Comment:
Do you know if any of these values are equivalent to not setting the ValidationDirective?

Reply:
I excluded that because, well, it depends. Here's an example from the k8s API docs

>fieldValidation instructs the server on how to handle objects in the request (POST/PUT/PATCH) containing unknown or duplicate fields, provided that the `ServerSideFieldValidation` feature gate is also enabled. Valid values are: - Ignore: This will ignore any unknown fields that are silently dropped from the object, and will ignore all but the last duplicate field that the decoder encounters. This is the default behavior prior to v1.23 and is the default behavior when the `ServerSideFieldValidation` feature gate is disabled. - Warn: This will send a warning via the standard warning response header for each unknown field that is dropped from the object, and for each duplicate field that is encountered. The request will still succeed if there are no other errors, and will only persist the last of any duplicate fields. This is the default when the `ServerSideFieldValidation` feature gate is enabled. - Strict: This will fail the request with a BadRequest error if any unknown fields would be dropped from the object, or if any duplicate fields are present. The error returned from the server will contain all unknown and duplicate fields encountered.

So it depends on which version of kubernetes the cluster is running, and if the feature gate is enabled. Not sure if we want to include all that detail as that may change.

Reply:
Ah, yeah, you're right. Looks like you've extracted the meaningful info.